src/deployment/deploy.py

```diff
@@ -300,7 +300,22 @@ def setup_rbac(self) -> None:
                 service_principal_type="Application",
                 app_id=app.app_id,
             )
-            client.service_principals.create(service_principal_params)
+            try:
+                client.service_principals.create(service_principal_params)
+            except GraphErrorException as err:
+                # work around timing issue when creating service principal
+                # https://github.com/Azure/azure-cli/issues/14767
+                if (
+                    "service principal being created must in the local tenant"
+                    not in str(err)
+                ):
+                    raise err
+                logging.warning(
+                    "creating service principal failed with an error that occurs "
+                    "due to AAD race conditions.  retrying after delay"
+                )
+                time.sleep(60)
```

Comment:
It looks like we're only retrying once, after a 1 min sleep.
1. Would it make more sense to retry N times with backoff?
2. If we fail with with the suspicious string in the `err` text, it would be good to emphasize that it may still be a resource creation race, but that we intentionally gave up after retrying.